app/core/engine.py

```diff
@@ -413,3 +415,63 @@ def execute_agent(
         return resp_debug
     else:
         return resp
+
+
+def clean_agent_memory(aid: str, thread_id: str = "", debug: bool = False):
+    """Clean an agent's memory with the given prompt and return response.
+
+    This function:
+    1. Cleans the agents skills data.
+    2. Cleans the thread skills data.
+    3. Cleans the graph checkpoint data.
+    4. Cleans the graph checkpoint_writes data.
+    5. Cleans the graph checkpoint_blobs data.
+
+    Args:
+        aid (str): Agent ID
+        thread_id (str): Thread ID for the agent memory cleanup
+        debug (bool): Enable debug mode
+
+    Returns:
+        str: Successful response message.
+    """
+    # get the agent from the database
+    with get_session() as db:
+        try:
+            AgentSkillData.clean_data(aid, db)
```

Comment:
split clean agent skill data and clean treads memory to 2 functions and 2 api, Most of the time, skill data actually doesn't need to be cleaned.